lib/model/dish_model.dart

```diff
@@ -6,38 +6,44 @@ class DishModel {
   String description;
   int calories;
   String imageUrl;
-  String dishType;
+  int dishTypeId;
+  String dishTypeName;
+  List<String> allergens;
 
   DishModel(
       {required this.title,
       this.description = "",
       this.calories = 0,
       this.imageUrl = "",
-      this.dishType = "",
-      this.id = -1});
+      this.dishTypeId = -1,
+      this.dishTypeName = "",
+      this.id = -1,
+      this.allergens = const [],
+      });
 
   Map<String, dynamic> toJson() {
     return {
       'title': title,
       'description': description,
       'calories': calories,
       'image': imageUrl,
-      'dishType': dishType,
+      'dishType': dishTypeName,
     };
   }
 
   static DishModel fromJson(Map<String, dynamic> input) {
     return DishModel(
-        id: input.containsKey("dish_id")
-          ? input["dish_id"]
+      id: input.containsKey("id")
+          ? input["id"]
           : throw Exception("No id provided"),
-        title: input.containsKey("title")
-            ? input["title"]
-            : throw Exception("No title provided"),
-        description:
-            input.containsKey("description") ? input["description"] : "",
-        calories: input.containsKey("calories") ? input["calories"] : 0,
-        imageUrl: input.containsKey("image") ? input["image"] : "",
-        dishType: input.containsKey("dish_type") ? input["dish_type"] : "",);
+      title: input.containsKey("title")
+          ? input["title"]
+          : throw Exception("No title provided"),
+      description: input.containsKey("description") ? input["description"] : "",
+      calories: input["calories"] ?? 0,
```

Comment:
Since calories is set by default to 0 on object creation, is the '?? 0' necessary?

Reply:
Yes. Because calories is nullable.